lib/src/container/unencapsulate.rs

```diff
@@ -289,19 +290,17 @@ pub(crate) async fn fetch_layer_decompress<'a>(
             })?;
             size = layer_blob.size;
             media_type = &layer_blob.media_type;
-            (blob, driver) = proxy
-                .get_blob(img, layer_blob.digest.as_str(), size as u64)
-                .await?;
+            (blob, driver) = proxy.get_blob(img, &layer_blob.digest, size).await?;
         }
         _ => {
             size = layer.size();
             media_type = layer.media_type();
-            (blob, driver) = proxy
-                .get_blob(img, layer.digest().as_str(), size as u64)
-                .await?;
+            (blob, driver) = proxy.get_blob(img, layer.digest(), size).await?;
         }
     };
 
+    let driver = async { driver.await.map_err(Into::into) };
```

Comment:
seems to be enough to accomodate the `thiserror` change, took me longer than I would care to admit to understand the compiler error before.

Reply:
I struggled a lot with this when I was learning Rust for sure. The thing that is subtle is that the `?` operator automatically also does `Into::into`, which means in cases where you're changing the error type (like this) but *not* using `?` (which happens in most cases) you suddenly have to think about these conversions.